### PR TITLE
Change invitations

### PR DESCRIPTION
**Pull request by kirknorthrop**

### CODE REVIEW

File: tickets/models.py [hunk 1]
```diff
@@ -121,17 +121,17 @@ def invitation(self):
...
+    token = models.CharField(max_length=24, unique=True)  # An index is automatically created since unique=True
     status = models.CharField(max_length=10, default='unclaimed')
 
     created_at = models.DateTimeField(auto_now_add=True)
     updated_at = models.DateTimeField(auto_now=True)
 
     class Manager(models.Manager):
         def create(self, **kwargs):
-            token = get_random_string(length=12)
+            token = get_random_string(length=24)
```
Comment (inglesp) [inline tickets/models.py]: I don't see the need to change this -- 12 characters is [71 bits of information](https://github.com/django/django/blob/master/django/utils/crypto.py#L54-L55).  While it's almost certainly a safe change to make, I'd like a little more convincing it's necessary.

File: orders/actions.py [hunk 2]
```diff
@@ -25,6 +25,7 @@ def process_stripe_charge(order, token):
     except IntegrityError:
         stripe_integration.refund_charge(charge.id)
         mark_order_as_errored_after_charge(order, charge.id)
+        logger.exception('An order faced an integrity error')
```
Comment (inglesp) [inline orders/actions.py]: Do we have anything on the `IntegrityError` we could log for added help in resolving the problem?  If so, perhaps we could record it on the `Order` object.

File: tickets/models.py (hunk 1 above)
Comment (kirknorthrop) [reply tickets/models.py]: It's a token only used in a URL, therefore the length is largely irrelevant as no-one has to manually quote it. Therefore it should be as long as makes reasonable sense. 

I would argue that UUIDs have excellent properties in this regard, and are in that "as long as makes reasonable sense" category, and I looked at the `shortuuid` library which uses base57 to make 22 character long representations of UUIDs. 

Not wishing to add another dependency, I just changed it to 24 characters, which is ~143 bits of entropy.

In short - no-one is having to quote this value, so there is no reason for it not to be as secure as possible.

File: orders/actions.py (hunk 2 above)
Comment (kirknorthrop) [reply orders/actions.py]: `logger.exception` logs the stack trace and reason for the integrity error - for example:
```django.db.utils.IntegrityError: duplicate key value violates unique constraint "tickets_ticketinvitation_email_addr_key"
DETAIL:  Key (email_addr)=(alice@example.com) already exists.```
I assume we've removed a significant source of potential `IntegrityError`s by making this change, and the logs appear in Heroku's dashboard, so therefore on the rare occasion they happen, it's probably just as easy to look at the logs.

File: tickets/models.py (hunk 1 above)
Comment (inglesp) [reply tickets/models.py]: I still don't understand why we need to change this (with 12 characters you're never going to get a collision, and nobody's going to bother trying to brute force a token) but don't really have a strong opinion.

File: orders/actions.py (hunk 2 above)
Comment (inglesp) [reply orders/actions.py]: Of course, yes.
